### PR TITLE
[QC][Hygon] Add INT8 varlen flash attention

### PR DESCRIPTION
**Pull request by July-h5kf3**
### PR Category

Operator

### Type of Change

New Feature, Performance Optimization

### Description

Adds `flash_attn_varlen_func_w8a8_int8` for Hygon DCU, dispatched by input dtype through the existing public `flaggems_vllm.flash_attn_varlen_func` entry. The implementation is grouped with the attention family in `src/flaggems_vllm/runtime/backend/_hygon/fused/attention.py`, with dedicated test and benchmark files.

Q/K/V inputs are INT8. QK uses INT8 dot products with INT32 accumulation; softmax and online output accumulation use FP32. Probabilities are quantized to 256 levels and PV uses INT8 dot with an explicit -128 zero-point correction. Grouped query heads share K/V loads, and mixed batches use a compact query grid to reduce empty programs. Kernel constants are `tl.constexpr` instances as required by Hygon Triton, and the block-size heuristics (BM/BN, warps, stages) are tuned for DCU - the benchmark's 64KB shared-memory budget rules out large BN, so long-query paths run BN=64 with BM=128 (compact) or BM=256 (non-compact).

Supports packed and paged KV, MHA/GQA/MQA, head dimensions 64/128, FP16/BF16 output, LSE, causal/local masks, ALiBi, and softcap. Tests cover these paths, block-varying and broadcast descales, strided inputs, and empty KV (115 passed). The benchmark reuses the existing FlagGems-vllm runner and all 18 workloads from vLLM's standard attention configuration. Operator metadata and CI test selection map the shared attention module to the dedicated tests and benchmark.

This is inference forward only. Dropout, attention-probability output, auxiliary queries, scheduler parameters, context parallelism, split-KV, and FA3 are unsupported.

Run locally with:

```bash
python3 -m pytest -v tests/test_flash_attn_varlen_func_w8a8_int8.py
python3 -m pytest -v -s benchmark/test_flash_attn_varlen_func_w8a8_int8.py --level core
```

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance

Measured on Hygon DCU BW1000 with PyTorch 2.4.1+das.opt2.dtk2504 and FlagTree Triton 3.6.0.

**Baseline: the existing FlagGems-vllm BF16 implementation of `flaggems_vllm.flash_attn_varlen_func`, running on the same device.** The benchmark's `_varlen_bf16_baseline` passes the original BF16 Q/K/V to the existing floating-point path in `ops/attention.py`. The INT8 implementation receives quantized versions of those inputs. Request lengths, page tables, head counts, head dimensions, and causal settings are identical. Hygon vLLM's attention does not provide a native INT8 baseline; the BF16 column is the FlagGems-vllm implementation, not native vLLM attention.

Timing uses the repository benchmark with the default warmup and measurement iterations. All GPU work within the attention call is included. Input quantization is excluded. Latencies below are arithmetic means of the recorded measurements; speedup is mean BF16 latency divided by mean INT8 latency.

Workloads follow [vLLM standard_attention.yaml](https://github.com/vllm-project/vllm/blob/main/benchmarks/attention_benchmarks/configs/standard_attention.yaml), with Hq/Hkv=32/8, D=128, page size=16, and causal=True. Each request group is `(request count, query length, total KV length)`; `+` denotes a mixed batch. Q shape is `[total_q, Hq, D]`; K/V shape is `[physical pages, page size, Hkv, D]`, including allocated unused pages.

|Request groups `(B, Sq, Sk)`|Q shape|K/V shape (each)|BF16 ms|INT8 ms|Speedup (BF16/INT8)|
|---|---|---|---:|---:|---:|
|`(1, 512, 512)`|`[512, 32, 128]`|`[32, 16, 8, 128]`|0.1830|0.0995|1.839x|
|`(1, 2048, 2048)`|`[2048, 32, 128]`|`[128, 16, 8, 128]`|0.9979|0.8206|1.216x|
|`(1, 4096, 4096)`|`[4096, 32, 128]`|`[256, 16, 8, 128]`|3.0479|2.6110|1.167x|
|`(1, 8192, 8192)`|`[8192, 32, 128]`|`[512, 16, 8, 128]`|10.3339|9.0860|1.137x|
|`(8, 1, 1024)`|`[8, 32, 128]`|`[512, 16, 8, 128]`|0.2926|0.1186|2.468x|
|`(16, 1, 2048)`|`[16, 32, 128]`|`[2048, 16, 8, 128]`|0.3638|0.2706|1.345x|
|`(32, 1, 1024)`|`[32, 32, 128]`|`[2048, 16, 8, 128]`|0.3547|0.2331|1.522x|
|`(64, 1, 4096)`|`[64, 32, 128]`|`[16384, 16, 8, 128]`|1.8342|1.3981|1.312x|
|`(2, 2048, 2048)` + `(8, 1, 1024)`|`[4104, 32, 128]`|`[1280, 16, 8, 128]`|3.0632|1.5901|1.926x|
|`(4, 1024, 1024)` + `(16, 1, 2048)`|`[4112, 32, 128]`|`[2560, 16, 8, 128]`|9.0406|1.2846|7.037x|
|`(2, 4096, 4096)` + `(32, 1, 1024)`|`[8224, 32, 128]`|`[8704, 16, 8, 128]`|12.5554|5.6397|2.226x|
|`(16, 2, 1024)`|`[32, 32, 128]`|`[1024, 16, 8, 128]`|0.4795|0.1477|3.247x|
|`(16, 4, 1024)`|`[64, 32, 128]`|`[1024, 16, 8, 128]`|0.4794|0.1478|3.242x|
|`(16, 8, 1024)`|`[128, 32, 128]`|`[1024, 16, 8, 128]`|0.4792|0.2106|2.276x|
|`(32, 4, 2048)`|`[128, 32, 128]`|`[4096, 16, 8, 128]`|1.6527|0.4211|3.925x|
|`(8, 8, 4096)`|`[64, 32, 128]`|`[2048, 16, 8, 128]`|1.0541|0.5010|2.104x|
|`(1, 1024, 2048)`|`[1024, 32, 128]`|`[128, 16, 8, 128]`|0.7400|0.6491|1.140x|
|`(2, 1024, 4096)`|`[2048, 32, 128]`|`[512, 16, 8, 128]`|2.4688|2.3040|1.072x|

**Overall: 2.233x arithmetic mean speedup over FlagGems-vllm BF16 across all 18 shapes.** Each shape has equal weight; the average is calculated from unrounded per-shape speedups. All **18/18** shapes are faster, with a minimum per-shape speedup of **1.072x**.

These results cover the listed single-attention workloads. They do not establish model-level accuracy, end-to-end serving throughput, or performance with substantially larger physical cache allocations. head_dim 64 is covered by functional tests but has no benchmark shape, so its heuristics are untuned, and very long pure prefill (8192+ KV) lands at ~1.1x with online-softmax elementwise work dominating - further gains there would need kernel-level changes.

### CODE REVIEW

**Comment by 0x45f** [comment]
please fix conflict

File: tests/test_flash_attn_varlen_func_w8a8_int8.py [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+
+
+@pytest.mark.parametrize("dim", [64, 128])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.bfloat16])
+@pytest.mark.parametrize("causal", [False, True])
+@pytest.mark.parametrize(
+    "qlens,klens",
+    [([1], [1]), ([17, 129], [33, 257]), ([131, 0, 3], [1, 0, 0]), ([128], [256])],
+)
+def test_packed(dim, dtype, causal, qlens, klens):
```
Comment (0x45f) [inline tests/test_flash_attn_varlen_func_w8a8_int8.py:198]: please add @pytest.mark.op_name

Comment (July-h5kf3) [reply tests/test_flash_attn_varlen_func_w8a8_int8.py:198]: Done

**Comment by 0x45f** [comment]
fix conflict

**Comment by July-h5kf3** [comment]
Done
